equation_generator.py

```diff
@@ -18,12 +18,16 @@ def generate_numbers_for_addition():
     """
     Generate two numbers that when added create an 8-character equation.
     Returns a tuple of (num1, num2, result)
-    
-    For addition, we want numbers that create 8 total characters
-    Format: NN+NN=NN (2+1+2+1+2 = 8 characters)
-
+    Format: NN+NN=NN  (2+1+2+1+2 = 8 characters)
     Example: (12, 34, 46) creates "12+34=46"
     """
+    while True:
+        num1 = random.randint(10, 99)
+        num2 = random.randint(10, 99)
+        result = num1 + num2
+        equation = f"{num1}+{num2}={result}"
+        if len(equation) == 8:
```

Comment:
Comparing the result mathematically would be faster (and more in line with best practices), rather than converting it to a string and checking the length